src/test/java/org/jabref/logic/journals/JournalAbbreviationRepositoryTest.java

```diff
@@ -134,4 +156,97 @@ void getFromFullName() {
     void getFromAbbreviatedName() {
         assertEquals(new Abbreviation("American Journal of Public Health", "Am. J. Public Health"), repository.get("Am. J. Public Health").get());
     }
+
+    @Test
+    void testAbbreviationsWithEscapedAmpersand() {
+        assertEquals(new Abbreviation("ACS Applied Materials & Interfaces", "ACS Appl. Mater. Interfaces"), repository.get("ACS Applied Materials & Interfaces").get());
+        assertEquals(new Abbreviation("ACS Applied Materials & Interfaces", "ACS Appl. Mater. Interfaces"), repository.get("ACS Applied Materials \\& Interfaces").get());
+        assertEquals(new Abbreviation("Antioxidants & Redox Signaling", "Antioxid. Redox Signaling"), repository.get("Antioxidants & Redox Signaling").get());
+        assertEquals(new Abbreviation("Antioxidants & Redox Signaling", "Antioxid. Redox Signaling"), repository.get("Antioxidants \\& Redox Signaling").get());
+
+        repository.addCustomAbbreviation(new Abbreviation("Long & Name", "L. N.", "LN"));
+        assertEquals(new Abbreviation("Long & Name", "L. N.", "LN"), repository.get("Long & Name").get());
+        assertEquals(new Abbreviation("Long & Name", "L. N.", "LN"), repository.get("Long \\& Name").get());
+    }
+
+    @Test
+    void testJournalAbbreviationWithEscapedAmpersand() throws IOException {
+        FieldWriterPreferences fieldWriterPreferences = new FieldWriterPreferences(true, List.of(StandardField.MONTH), new FieldContentFormatterPreferences());
+        BibEntryWriter bibEntryWriter = new BibEntryWriter(new FieldWriter(fieldWriterPreferences), new BibEntryTypesManager());
+        StringWriter stringWriter = new StringWriter();
+        BibWriter bibWriter = new BibWriter(stringWriter, OS.NEWLINE);
+
+        BibDatabase bibDatabase = new BibDatabase();
+        JournalAbbreviationRepository journalAbbreviationRepository = JournalAbbreviationLoader.loadBuiltInRepository();
+        UndoableAbbreviator undoableAbbreviator = new UndoableAbbreviator(journalAbbreviationRepository, AbbreviationType.DEFAULT);
+
+        BibEntry entryWithEscapedAmpersandInJournal = new BibEntry(StandardEntryType.Article);
+        entryWithEscapedAmpersandInJournal.setField(StandardField.JOURNAL, "ACS Applied Materials \\& Interfaces");
+
+        // @formatter:off
+        String expectedAbbreviatedJournalEntry = """
+                @Article{,
+                  journal = {ACS Appl. Mater. Interfaces},
+                }
+                """.replaceAll("\n", OS.NEWLINE);
+        // @formatter:on
+
+        undoableAbbreviator.abbreviate(bibDatabase, entryWithEscapedAmpersandInJournal, StandardField.JOURNAL, new CompoundEdit());
+        bibEntryWriter.write(entryWithEscapedAmpersandInJournal, bibWriter, BibDatabaseMode.BIBLATEX);
+        assertEquals(expectedAbbreviatedJournalEntry, stringWriter.toString());
```

Comment:
Comparison on a String bases should only be used for serialization tests.

In your case, do

```suggestion
        undoableAbbreviator.abbreviate(bibDatabase, entryWithEscapedAmpersandInJournal, StandardField.JOURNAL, new CompoundEdit());
        BibEntry expectedAbbreviatedJournalEntry = new BibEntry(StandardEntryType.Article)
                .withField(StandardField.JOURNAL, "ACS Appl. Mater. Interfaces");
        assertEquals(expectedAbbreviatedJournalEntry , entryWithEscapedAmpersandInJournal);
```